applications/DigitalTwinApplication/tests/test_system_identification.py

```diff
@@ -0,0 +1,44 @@
+import numpy
+import KratosMultiphysics as Kratos
+import KratosMultiphysics.KratosUnittest as UnitTest
+from KratosMultiphysics.StructuralMechanicsApplication.structural_mechanics_analysis import StructuralMechanicsAnalysis
+from KratosMultiphysics.OptimizationApplication.optimization_analysis import OptimizationAnalysis
+from KratosMultiphysics.kratos_utilities import DeleteFileIfExisting
+from KratosMultiphysics.compare_two_files_check_process import CompareTwoFilesCheckProcess
+
+class TestSystemIdentification(UnitTest.TestCase):
+    def test_DamagedSystem(self):
+        self.addCleanup(DeleteFileIfExisting, "auxiliary_files/damaged_problem/measured_data.csv")
+
+        model = Kratos.Model()
+        with open("auxiliary_files/damaged_problem/damaged_project_parameters.json", "r") as file_input:
+            params = Kratos.Parameters(file_input.read())
+        analysis = StructuralMechanicsAnalysis(model, params)
+        analysis.Run()
+
+        data = numpy.loadtxt("auxiliary_files/damaged_problem/measured_data.csv", comments="#", usecols=[0,3,4,5,6], delimiter=",")
+        ref_data = numpy.loadtxt("auxiliary_files/damaged_problem/measured_data_ref.csv", comments="#", usecols=[0,3,4,5,6], delimiter=",")
+        self.assertAlmostEqual(numpy.linalg.norm(data-ref_data), 0.0, 9)
```

Comment:
do you know about `np.testing.assert_allclose`?
Or to make it compatible with unittest you can use `self.assertTrue(np.allclose(a, b)`

Or self.assertMatrixAlmostEqual

Reply:
done :)